pkg/pulumiyaml/analyser.go

```diff
@@ -532,9 +534,15 @@ func hasValidEnumValue(from ast.Expr, to []*schema.Enum) *notAssignable {
 	}
 }
 
+var disableTypeChecking = cmdutil.IsTruthy(os.Getenv("PULUMI_YAML_DISABLE_TYPE_CHECKING"))
+
 // Provides an appropriate diagnostic message if it is illegal to assign `from`
 // to `to`.
 func (tc *typeCache) assertTypeAssignable(ctx *evalContext, from ast.Expr, to schema.Type) {
+	if disableTypeChecking {
+		return
+	}
```

Comment:
I'd probably print a big warning here that this is not something that users should set, the provider may panic, etc.

Reply:
What's the accepted way to print a warning in the language plugin?

I tried:
```golang
ctx.addWarnDiag(
	from.Syntax().Syntax().Range(),
	"Running with type checking disabled. This is not recommended for production use.",
	"Type checking is disabled. This is a test feature and should not be used in production.",
)
```

but that didn't work

Reply:
That's what I would recommend. What does "didn't work" mean? Did it just not do anything?

Reply:
Yeah, it didn't print anything

Reply:
Hmm. That's concerning. I wouldn't block on this.

Reply:
Adding the warning is probably still a good idea, and opening an issue.  We have other warnings below that we should probably also fix if this doesn't work.

Reply:
Opened https://github.com/pulumi/pulumi-yaml/issues/638